src/hotspot/cpu/x86/stubGenerator_x86_64_fmod.cpp

```diff
@@ -0,0 +1,546 @@
+    0x7FF0000000000000ULL,    // Inf
+};
+
+ATTRIBUTE_ALIGNED(32) static const uint64_t CONST_e307[] = {
+    0x7FE0000000000000ULL
+};
+
+address StubGenerator::generate_libmFmod() {
+  StubCodeMark mark(this, "StubRoutines", "libmFmod");
+  address start = __ pc();
+  __ enter(); // required for proper stackwalking of RuntimeStub frame
+
+  if (VM_Version::supports_avx512vlbwdq()) {     // AVX512 version
+
+  Label L_12ca, L_1280, L_115a, L_1271, L_1268, L_1227, L_1231, L_11f4, L_128a, L_1237, L_12bf, L_1290, L_exit;
+
+// #include <ia32intrin.h>
+// #include <emmintrin.h>
+// #pragma float_control(precise, on)
+
+// #define UINT32 unsigned int
+// #define SINT32 int
+// #define UINT64 unsigned __int64
+// #define SINT64 __int64
+
+// #define DP_FMA(a, b, c)    __fence(_mm_cvtsd_f64(_mm_fmadd_sd(_mm_set_sd(a), _mm_set_sd(b), _mm_set_sd(c))))
+// #define DP_FMA_RN(a, b, c)    _mm_cvtsd_f64(_mm_fmadd_round_sd(_mm_set_sd(a), _mm_set_sd(b), _mm_set_sd(c), (_MM_FROUND_TO_NEAREST_INT | _MM_FROUND_NO_EXC)))
+// #define DP_FMA_RZ(a, b, c) __fence(_mm_cvtsd_f64(_mm_fmadd_round_sd(_mm_set_sd(a), _mm_set_sd(b), _mm_set_sd(c), (_MM_FROUND_TO_ZERO | _MM_FROUND_NO_EXC))))
+
+// #define DP_ROUND_RZ(a)   _mm_cvtsd_f64(_mm_roundscale_sd(_mm_setzero_pd(), _mm_set_sd(a), (_MM_FROUND_TO_ZERO | _MM_FROUND_NO_EXC)))
+
+// #define DP_CONST(C)    _castu64_f64(0x##C##ull)
+// #define DP_AND(X, Y)   _mm_cvtsd_f64(_mm_and_pd(_mm_set_sd(X), _mm_set_sd(Y)))
+// #define DP_XOR(X, Y)   _mm_cvtsd_f64(_mm_xor_pd(_mm_set_sd(X), _mm_set_sd(Y)))
+// #define DP_OR(X, Y)    _mm_cvtsd_f64(_mm_or_pd(_mm_set_sd(X), _mm_set_sd(Y)))
+// #define DP_DIV_RZ(a, b) __fence(_mm_cvtsd_f64(_mm_div_round_sd(_mm_set_sd(a), _mm_set_sd(b), (_MM_FROUND_TO_ZERO | _MM_FROUND_NO_EXC))))
+// #define DP_FNMA(a, b, c)    __fence(_mm_cvtsd_f64(_mm_fnmadd_sd(_mm_set_sd(a), _mm_set_sd(b), _mm_set_sd(c))))
+// #define DP_FNMA_RZ(a, b, c) __fence(_mm_cvtsd_f64(_mm_fnmadd_round_sd(_mm_set_sd(a), _mm_set_sd(b), _mm_set_sd(c), (_MM_FROUND_TO_ZERO | _MM_FROUND_NO_EXC))))
+
+// #define D2L(x)  _mm_castpd_si128(x)
+// // transfer highest 32 bits (of low 64b) to GPR
+// #define TRANSFER_HIGH_INT32(X)   _mm_extract_epi32(D2L(_mm_set_sd(X)), 1)
+
+
+// double fmod(double x, double y)
+// {
+  __ subptr(rsp, 0x28);
+  __ mov64(rax, 0x7fffffffffffffff);
+  __ movapd(xmm20, xmm0);
+  __ movapd(xmm3, xmm1);
+// double a, b, sgn_a, q, bs, bs2;
+// unsigned eq;
+
+//     // |x|, |y|
+//     a = DP_AND(x, DP_CONST(7fffffffffffffff));
+  __ vxorpd(xmm18, xmm18, xmm18, Assembler::AVX_512bit);
+  __ vmovsd(xmm5, xmm18, xmm20);
+  __ movq(xmm17, rax);
+  __ vandpd(xmm0, xmm5, xmm17, Assembler::AVX_512bit);
+//     b = DP_AND(y, DP_CONST(7fffffffffffffff));
+  __ vandpd(xmm1, xmm3, xmm17, Assembler::AVX_512bit);
+//   // sign(x)
+//   sgn_a = DP_XOR(x, a);
+  __ vmovsd(xmm17, xmm18, xmm0);
+  __ vxorpd(xmm16, xmm5, xmm17, Assembler::AVX_512bit);
+
+//   q = DP_DIV_RZ(a, b);
+  __ vmovsd(xmm5, xmm18, xmm1);
```

Comment:
This and other usage of vmovsd with blending two registers could be avoided.

Reply:
I don't know what you mean.  Can you elaborate please?